lib/es-models/item.js

```diff
@@ -0,0 +1,372 @@
+    const item = this.item
+    if (item.varField(shelfMarkMarc, ['h'])) {
+      callnum = item.varField(shelfMarkMarc, shelfMarkSubfields)[0]
+    }
+    // If not found in var field
+    if (!callnum) callnum = item.callNumber
+    // If not yet found, look in 945 $g
+    if (!callnum && item.varField('945', ['g']).length > 0) callnum = item.varField('945', ['g'])[0]
+    // If not found, default to callnum from bib
+    if (!callnum) {
+      const bibCallNum = this.bib._callNum()
+      if (bibCallNum && bibCallNum.value) callnum = bibCallNum
+    }
+    // Sierra seems to put these '|h' prefixes on callnumbers; strip 'em
+    if (callnum && typeof callnum === 'object' && callnum.value) {
+      callnum = callnum.value
+    }
+    if (callnum) callnum = callnum.replace(/^\|h/, '')
+    return callnum
+  }
+
+  // ToDO: getting callnum from bib
+  shelfMark () {
+    let callnum = this._callNum()
+    // If not found in var field
+    if (callnum) {
+      // Pull callnumber suffix from fieldTag v if present
+      const callnumSuffix = this.item.fieldTag('v')
+
+      if (callnumSuffix && callnumSuffix.length) {
+        callnum += ' ' + callnumSuffix[0].value
+      }
+
+      // Finally store the complete shelfMark data
+      return [callnum]
+    }
+  }
+
+  shelfMark_sort () {
+    return this._sortableShelfMark()[0]
+  }
+
+  physicalLocation () {
+    return [this._callNum()]
+  }
+
+  // trim ? coreObjectsMappingByCode
+  status () {
+    // Availability
+    let status = null
+    const item = this.item
+    // Look for a truthy status.code
+    if (item.isNyplRecord() && item.status && item.status.code && item.status.code.trim()) {
+      status = { path: 'status' }
+      if (item.status.code === '-') {
+        // Code '-' matches both Available and Loaned (!!!) so, check duedate
+        status.id = item.status.duedate ? 'co' : 'a'
+      } else {
+        status.code = item.status.code
+      }
+    } else if (item.varField('876', ['j']).length > 0) {
+      // Mainly serving recap, mapp string statuses:
+      status = { path: '876 $j' }
+      const val = item.varField('876', ['j'])[0].toLowerCase()
+      switch (val) {
+        case 'available': status.id = 'a'; break
+        case 'not available': status.id = 'na'; break
+        case 'loaned': status.id = 'r'; break
+      }
+    }
+    if (status) {
+      // We either know the status id or the status code
+      // Find the right entry in our status mapping based on which we identified:
+      const statusMapped = status.id
+        ? statusMapping[status.id]
+        : coreObjectsMappingByCode(statusMapping, status.code)
+      if (statusMapped) {
+        const statusId = statusMapped.id.split(':').pop()
+        return [{ id: `status:${statusId}`, label: statusMapped.label.trim() }]
+      } else logger.error('could not find status for:', status)
+    }
+  }
+
+  // does match just mean == ?
+  aeonUrl () {
+    const baseAeonUrl = this.bib._aeonUrl()
```

Comment:
The EsBib method is `_aeonUrls()` and it returns null or an array. So maybe something like `const baseAeonUrl = (this.bib._aeonUrls() || []).shift()` ? (The [DHI implementation](https://github.com/NYPL/discovery-api-indexer/blob/31f03b583e3d2ddaef6c30235889ed63724a4182/lib/serializers/resource-item-serializer.js#L114-L115) just uses the first Aeon URL found in the bib, so `.shift()` is fine)

Reply:
Sounds good

Reply:
The EsBib method is plural - `_aeonUrls()`

Reply:
Whoops - I see the change now. Thanks